src/tests/nativeaot/GenerateUnmanagedEntryPoints/GenerateUnmanagedEntryPoints.cpp

```diff
@@ -0,0 +1,61 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifdef TARGET_WINDOWS
+#include "windows.h"
+#else
+#include "dlfcn.h"
+#endif
+#include "stdio.h"
+#include "string.h"
+
+#ifndef TARGET_WINDOWS
+#define __stdcall
+#endif
+
+// typedef for shared lib exported methods
+typedef void(__stdcall *fptr)();
+
+#ifdef TARGET_WINDOWS
+int __cdecl main()
+#else
+int main(int argc, char* argv[])
+#endif
+{
+#ifdef TARGET_WINDOWS
+    HINSTANCE handle = LoadLibrary("GenerateUnmanagedEntryPoints.dll");
```

Comment:
Can the test be a fully managed test without any C++? `dlopen` and `dlsym` can be done from C# just fine, without any ifdefs.

Reply:
The test can even be a single executable - you set `IlcExportUnmanagedEntrypoints>true</IlcExportUnmanagedEntrypoints>` to export the entrypoints. It will avoid `CLRTestBatchPreCommands` that tend to be pain to deal with.